src/aeso_fcode_to_fate.erl

```diff
@@ -751,6 +734,66 @@ push(A) -> {'STORE', ?a, A}.
 tuple(0) -> push(?i({tuple, {}}));
 tuple(N) -> aeb_fate_ops:tuple(?a, N).
 
+dbg_loc(Env, Ann) ->
+    case proplists:get_value(debug_info, Env#env.options, false) of
+        false -> [];
+        true  ->
+            File = case proplists:get_value(file, Ann) of
+                       no_file -> "";
+                       F       -> F
+                   end,
+            Line = proplists:get_value(line, Ann),
+            case Line of
+                undefined -> [];
+                _         -> [{'DBG_LOC', {immediate, File}, {immediate, Line}}]
+            end
+    end.
+
+dbg_scoped_vars(_Env, [], SCode) ->
+    SCode;
+dbg_scoped_vars(Env, [Var | Rest], SCode) ->
+    dbg_scoped_vars(Env, Rest, dbg_scoped_var(Env, Var, SCode)).
+
+dbg_scoped_var(Env = #env{saved_fresh_names = SavedFreshNames}, Var, SCode) ->
+    case proplists:get_value(debug_info, Env#env.options, false) of
+        false -> SCode;
+        true  ->
+            case maps:get(Var, SavedFreshNames, Var) of
+                "%" ++ _ -> SCode;
+                "_"      -> SCode;
+                VarName  ->
+                    Register = lookup_var(Env, Var),
+                    Def      = [{'DBG_DEF',   {immediate, VarName}, Register}],
+                    Undef    = [{'DBG_UNDEF', {immediate, VarName}, Register}],
+                    Def ++ dbg_undef(Undef, SCode)
+            end
+    end.
+
+dbg_undef(_Undef, missing) ->
+    missing;
+dbg_undef(Undef, loop) ->
+    [Undef, loop];
+dbg_undef(Undef, switch_body) ->
+    [switch_body, Undef];
+dbg_undef(Undef, {switch, Arg, Type, Alts, Catch}) ->
+    NewAlts   = [ dbg_undef(Undef, Alt) || Alt <- Alts ],
+    NewCatch  = dbg_undef(Undef, Catch),
+    NewSwitch = {switch, Arg, Type, NewAlts, NewCatch},
+    NewSwitch;
+dbg_undef(Undef, SCode) when is_list(SCode) ->
+    lists:droplast(SCode) ++ [dbg_undef(Undef, lists:last(SCode))];
+dbg_undef(Undef, SCode) when is_tuple(SCode); is_atom(SCode) ->
+    [Mnemonic | _] =
+        case is_tuple(SCode) of
+            true  -> tuple_to_list(SCode);
+            false -> [SCode]
+        end,
+    Op = aeb_fate_opcodes:m_to_op(Mnemonic),
+    case aeb_fate_opcodes:end_bb(Op) of
+        true  -> [Undef, SCode];
```

Comment:
In which case would you see something like `RETURNR {var, 1}`? The return result is pushed to the stack and `RETURN` is used instead of `RETURNR` most of the times.

Can you point to an example that would produce the case that you're talking about?

Reply:
```
  entrypoint f(x)  : int =
    x
```
compiles to
```
FUNCTION f( integer) : integer
  ;; BB : 0
          RETURNR arg0
```
-----------
I just want to be sure if we cover such cases properly. Even though we may not use `RETURNR` extensively now, we might use it later for whatever reason --- let's be prepared for such possibility.

Reply:
I'm turning off all optimizations including swap_push optimization for debugging compilation, so they won't happen, and the generated fate code won't have a `RETURNR arg0`. But it's something to consider in the future.

Reply:
If you insist on not covering it now, make at least an issue for that